screen-orientation/window-orientation.html

```diff
@@ -0,0 +1,182 @@
+
+promise_test(async (t) => {
+  t.add_cleanup(makeCleanup());
+  await test_driver.bless("request full screen");
+  await document.documentElement.requestFullscreen();
+
+  const initialOrientation = window.orientation;
+  const targetOrientation = getOppositeOrientation();
+
+  const eventPromise = new Promise(resolve => {
+    function handleOrientationChange() {
+      window.removeEventListener("orientationchange", handleOrientationChange);
+      resolve();
+    }
+    window.addEventListener("orientationchange", handleOrientationChange);
+  });
+
+  // Lock the orientation
+  await screen.orientation.lock(targetOrientation);
+
+  // Wait for the orientationchange event
+  await eventPromise;
+
+  // Verify the orientation value is still valid
+  const newOrientation = window.orientation;
+  assert_true(
+    newOrientation === -90 || newOrientation === 0 || newOrientation === 90 || newOrientation === 180,
+    "window.orientation should be a valid value after orientation change"
+  );
+}, "orientationchange event should fire when orientation is changed programmatically");
+
+test(() => {
+  // Test that orientation values map correctly to screen orientation angles
+  const orientation = window.orientation;
+  const screenAngle = screen.orientation.angle;
+
+  let expectedOrientation;
+  if (screenAngle < 180) {
+    expectedOrientation = screenAngle;
+  } else if (screenAngle === 180) {
+    // User agents may or may not support 180
+    expectedOrientation = screenAngle; // or 0 if not supported
+  } else if (screenAngle > 180) {
+    expectedOrientation = screenAngle - 360;
+  }
+
+  // Allow for the fact that some user agents might not support 180
+  if (screenAngle === 180 && orientation === 0) {
+    assert_true(true, "User agent correctly maps unsupported 180° to 0°");
+  } else {
+    assert_equals(orientation, expectedOrientation,
+      `window.orientation (${orientation}) should match expected mapping of screen.orientation.angle (${screenAngle})`);
+  }
+}, "window.orientation should correctly map screen.orientation.angle values");
+
+test(() => {
+  // Test the specific constraints from the spec
+  const orientation = window.orientation;
+
+  // Must support -90, 0, 90
+  if (orientation === -90 || orientation === 0 || orientation === 90) {
+    assert_true(true, "Orientation is one of the required supported values");
+  }
+  // May optionally support 180
+  else if (orientation === 180) {
+    assert_true(true, "Orientation is the optional 180° value");
+  }
+  else {
+    assert_unreached(`Invalid orientation value: ${orientation}`);
+  }
+}, "window.orientation should only return standardized values");
+
+promise_test(async (t) => {
+  t.add_cleanup(makeCleanup());
+  await test_driver.bless("request full screen");
+  await document.documentElement.requestFullscreen();
+
+  // Test different orientation locks and verify window.orientation values
+  const testCases = [
+    { lock: "portrait", expectedValues: [0, 180] },
```

Comment:
If this is mobile device, default is "portrait". But if Windows tablet, default may be "landscape" (Firefox supports orientation APIs on Window tablet). So it is incorrect for all situation.